fmriprep/workflows/base.py

```diff
@@ -237,41 +250,72 @@ def basic_wf(subject_data, settings, name='fMRI_prep'):
+                                      'inputnode.t1_2_mni_forward_transform')]),
+        (hmcwf, epi_2_t1, [('inputnode.epi', 'inputnode.name_source'),
+                           ('outputnode.epi_split', 'inputnode.epi_split')]),
         (hmcwf, confounds_wf, [
             ('outputnode.movpar_file', 'inputnode.movpar_file'),
             ('outputnode.motion_confounds_file', 'inputnode.motion_confounds_file'),
             ('inputnode.epi', 'inputnode.source_file')]),
-        (epi_2_t1, confounds_wf, [('outputnode.epi_t1', 'inputnode.fmri_file'),
-                                  ('outputnode.epi_mask_t1', 'inputnode.epi_mask')]),
-
-        (epi_2_t1, epi_mni_trans_wf, [('outputnode.itk_epi_to_t1', 'inputnode.itk_epi_to_t1')]),
         (hmcwf, epi_mni_trans_wf, [('inputnode.epi', 'inputnode.name_source'),
-                                   ('outputnode.xforms', 'inputnode.hmc_xforms'),
-                                   ('outputnode.epi_mask', 'inputnode.epi_mask'),
                                    ('outputnode.epi_split', 'inputnode.epi_split')]),
-        (t1w_pre, epi_mni_trans_wf, [('outputnode.bias_corrected_t1', 'inputnode.t1'),
-                                     ('outputnode.t1_2_mni_forward_transform',
-                                      'inputnode.t1_2_mni_forward_transform')])
     ])
 
-    if settings['freesurfer']:
+    if fmap_est is None:
+        workflow.connect([
+            (hmcwf, epi_2_t1, [('outputnode.epi_mean', 'inputnode.ref_epi'),
+                               ('outputnode.xforms', 'inputnode.hmc_xforms'),
+                               ('outputnode.epi_mask', 'inputnode.ref_epi_mask')]),
+            (hmcwf, epi_mni_trans_wf, [('outputnode.xforms', 'inputnode.hmc_xforms'),
+                                       ('outputnode.epi_mask', 'inputnode.epi_mask')]),
+        ])
+    else:
+        workflow.connect([
+            (hmcwf, unwarp, [('inputnode.epi', 'inputnode.name_source'),
+                             ('outputnode.epi_split', 'inputnode.in_split'),
+                             ('outputnode.epi_mean', 'inputnode.in_reference'),
+                             ('outputnode.xforms', 'inputnode.xforms')]),
+            (fmap_est, unwarp, [('outputnode.fmap', 'inputnode.fmap'),
+                                ('outputnode.fmap_ref', 'inputnode.fmap_ref'),
+                                ('outputnode.fmap_mask', 'inputnode.fmap_mask')]),
+            (unwarp, epi_2_t1, [('outputnode.out_reference', 'inputnode.ref_epi'),
+                                ('outputnode.out_warps', 'inputnode.hmc_xforms'),
+                                ('outputnode.out_mask', 'inputnode.ref_epi_mask')]),
+            (unwarp, epi_mni_trans_wf, [('outputnode.out_warps', 'inputnode.hmc_xforms'),
+                                        ('outputnode.out_mask', 'inputnode.epi_mask')])
+        ])
+
+        # Report on EPI correction
+        epireport = epi_preproc_report(settings=settings)
```

Comment:
This seems like a short workflow that is newer reused. All other reportlet datasinks were defined in the main workflow directly - shouldn't we do the same for consistency?